cmd/zapi/cmd/get/get.go

```diff
@@ -0,0 +1,184 @@
+	f.BoolVar(&c.ShowFields, "F", false, "display field names in text output")
+	f.BoolVar(&c.EpochDates, "E", false, "display epoch timestamps in text output")
+	f.BoolVar(&c.wire, "w", false, "dump what's on the wire")
+	return c, nil
+}
+
+func (c *Command) Run(args []string) error {
+	// XXX For now only allow non-wire searches to run with zng response
+	// encoding. It shouldn't be difficult to allow this for all supported
+	// response encodings but KISS for now.
+	if !c.wire && c.protocol != "zng" {
+		return errors.New("only zng protocol allowed for non-wire searches")
+	}
+	expr := "*"
+	if len(args) > 0 {
+		expr = strings.Join(args, " ")
+	}
+	client := c.Client()
+	id, err := c.SpaceID()
+	if err != nil {
+		return err
+	}
+	req, err := parseExpr(id, expr)
+	if err != nil {
+		return fmt.Errorf("parse error: %s", err)
+	}
+	req.Span, err = fromTo(c.Context(), client, id)
+	if err != nil {
+		return fmt.Errorf("%s: no such space: %s", c.Spacename, err)
+	}
+	if req.Span.Dur == 0 {
+		// this can happen for empty spaces
+		return fmt.Errorf("%s: space is empty", c.Spacename)
+	}
+	params := map[string]string{"format": c.protocol}
+	r, err := client.SearchRaw(c.Context(), *req, params)
+	if err != nil {
+		return fmt.Errorf("search error: %w", err)
+	}
+	defer r.Close()
+	if c.wire {
+		return c.runWireSearch(r)
+	}
+	writer, err := openOutput(c.dir, c.outputFile, c.WriterFlags)
+	if err != nil {
+		return err
+	}
+	stream := api.NewZngSearch(r) // XXX fix this to match specified format
+	stream.SetOnCtrl(c.handleControl)
+	if err := zbuf.Copy(zbuf.NopFlusher(writer), stream); err != nil {
+		writer.Close()
+		if c.Context().Err() != nil {
+			fmt.Fprintln(os.Stderr, "search aborted")
+			return nil
+		}
+		return err
+	}
+	return writer.Close()
+}
+
+// parseExpr creates an api.SearchRequest to be used with the client.
+func parseExpr(spaceID api.SpaceID, expr string) (*api.SearchRequest, error) {
+	search, err := zql.ParseProc(expr)
+	if err != nil {
+		return nil, err
+	}
+	proc, err := json.Marshal(search)
+	if err != nil {
+		return nil, err
+	}
+	return &api.SearchRequest{
+		Space: spaceID,
+		Proc:  proc,
+	}, nil
+}
+
+func (c *Command) handleControl(ctrl interface{}) {
+	switch ctrl := ctrl.(type) {
+	case *api.SearchStats:
+		if c.stats {
+			c.final = ctrl
+		}
+	case *api.SearchWarning:
+		if c.warnings {
+			for _, w := range ctrl.Warning {
+				fmt.Fprintln(os.Stderr, w)
+			}
+		}
+	}
+}
+
+func (c *Command) runWireSearch(r io.Reader) error {
+	w := os.Stdout
+	if c.outputFile != "" {
+		var err error
+		flags := os.O_WRONLY | os.O_CREATE | os.O_TRUNC
+		file, err := fs.OpenFile(c.outputFile, flags, 0600)
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+		w = file
+	}
+	_, err := io.Copy(w, r)
+	return err
+}
+
+func fromTo(ctx context.Context, client *api.Connection, id api.SpaceID) (nano.Span, error) {
+	//XXX for now we run the search over the entire space
```

Comment:
I think the ability to specify the time range should go in this PR.

Reply:
I can do that. How would you imagine this working? Just simple `-from` / `-to` args that default to min/max time if not set? Do we want to support relative time (e.g `-last 5mins`)  How crazy do we want to get with parsing human readable time formats (I suppose I could search for some suitable library that does this)?

Reply:
My suggestion would be to start small: `-from` and/or `-to` that are come from calling `time.Parse` with `RFC3339Nano`.

Reply:
Sounds good